webapp/lib/model.dart

```diff
@@ -6,6 +6,16 @@ export 'model.g.dart' hide
   MessageDirection_fromStringOverride,
   TagType_fromStringOverride;
 
+extension UserConfigurationUtil on g.UserConfiguration {
+  g.UserConfiguration operator +(g.UserConfiguration other) =>
```

Comment:
This is a very good point, I hadn't thought about maintaining commutativity. I like your solution so I've rewritten the code to use `applyDefaults` instead of `+`, thank you!